pkg/controller/controllername/controllername_test.go

```diff
@@ -0,0 +1,80 @@
+package controllername
+
+import (
+	"github.com/stretchr/testify/require"
+	"regexp"
+	"testing"
+)
+
+func TestMetricsName(t *testing.T) {
+
+	cn1 := NewControllerName([]string{"SomeFakeControllerName"})
+	cn2 := NewControllerName([]string{"Some", "Controller", "Name"})
+	cn3 := NewControllerName([]string{" SomeName", "Entered  ", "poorly"})
+
+	metricName1 := cn1.MetricsName()
+	metricName2 := cn2.MetricsName()
+	metricName3 := cn3.MetricsName()
+
+	require.True(t, isPrometheusBestPracticeName(metricName1))
+	require.True(t, isPrometheusBestPracticeName(metricName2))
+	require.True(t, isPrometheusBestPracticeName(metricName3))
+}
+
+func TestLoggerName(t *testing.T) {
+	cn1 := NewControllerName([]string{"SomeFakeControllerName"})
+	cn2 := NewControllerName([]string{"Some", "Controller", "Name"})
+	cn3 := NewControllerName([]string{" SomeName", "Entered  ", "poorly"})
+
+	loggerName1 := cn1.LoggerName()
+	loggerName2 := cn2.LoggerName()
+	loggerName3 := cn3.LoggerName()
+
+	require.True(t, isBestPracticeLoggerName(loggerName1))
+	require.True(t, isBestPracticeLoggerName(loggerName2))
+	require.True(t, isBestPracticeLoggerName(loggerName3))
+}
+
+func TestIsPrometheusBestPracticeName(t *testing.T) {
```

Comment:
valuable?

Reply:
i removed both. They were valuable at one point, but we change from the clean implementation to the New implementation.